### PR TITLE
Kinno 642 reciever support transaction history

### PR DESCRIPTION
**Pull request by shayb**
#### Main purpose:
add ability to receiver to show transaction history
#### Technical description:
(What was changed/added)
#### Dilemmas you faced with? 
(Not a must)
#### Tests:
(Main tests you added/changed)
#### Documentation:
(Source/readme.md)

### CODE REVIEW

File: core/src/main/java/com/kin/ecosystem/core/data/auth/AuthRepository.java [hunk 1]
```diff
@@ -147,11 +147,14 @@ public AuthToken getAuthTokenSync() {
...
+		if(localData.isLoggedIn()) {
+			if (cachedAuthToken != null) {
+				return isAuthTokenExpired(cachedAuthToken);
+			}
+			final AuthToken token = localData.getAuthTokenSync();
+			if (token != null) {
+				return isAuthTokenExpired(token);
+			}
 		}
         return false;
```
Comment (shayb) [inline core/src/main/java/com/kin/ecosystem/core/data/auth/AuthRepository.java]: maybe should be true ?

File: sdk/src/main/java/com/kin/ecosystem/transfer/view/AccountInfoActivity.java [hunk 2]
```diff
@@ -40,11 +41,29 @@ public String getData() throws AccountInfoException {
...
+                    IncomingTransfer payload = new IncomingTransfer().appId(senderAppId).memo(memo).title(title).description(description).walletAddress("");
+                    if (orderRepository != null) {
+                        try {
+                            orderRepository.createIncomingTransferOrderSync(payload);
+                        } catch (KinEcosystemException e) {
+                            e.printStackTrace();
+                        }
+                    }
+                }
+            });
```
Comment (shayb) [inline sdk/src/main/java/com/kin/ecosystem/transfer/view/AccountInfoActivity.java]: Do it async from the repo itself.

**Review by OrenZak — APPROVED** [review]
Please check after the changes 👍

File: core/src/main/java/com/kin/ecosystem/core/data/auth/AuthRepository.java (hunk 1 above)
Comment (shayb) [reply core/src/main/java/com/kin/ecosystem/core/data/auth/AuthRepository.java]: @OrenZak  no this is correct - only check validity of the token when user is logged in

Comment (OrenZak) [reply core/src/main/java/com/kin/ecosystem/core/data/auth/AuthRepository.java]: but returning `false` at the end means that the authToken is valid and not expired but the user is not logged in at all so it's not make sense.

File: sdk/src/main/java/com/kin/ecosystem/transfer/view/AccountInfoActivity.java [hunk 3]
```diff
@@ -40,11 +41,29 @@ public String getData() throws AccountInfoException {
...
+                @Override
+                public void run() {
+                    String title = "Received Kin";
+                    String description = "From " + senderAppName;
+                    final OrderRepository orderRepository = OrderRepository.getInstance();
+                    IncomingTransfer payload = new IncomingTransfer().appId(senderAppId).memo(memo).title(title).description(description).walletAddress("");
+                    if (orderRepository != null) {
+                        try {
+                            orderRepository.createIncomingTransferOrderSync(payload);
+                        } catch (KinEcosystemException e) {
```
Comment (OrenZak) [inline sdk/src/main/java/com/kin/ecosystem/transfer/view/AccountInfoActivity.java]: why it still sync?

File: core/src/main/java/com/kin/ecosystem/core/data/auth/AuthRepository.java (hunk 1 above)
Comment (shayb) [reply core/src/main/java/com/kin/ecosystem/core/data/auth/AuthRepository.java]: i will throw an exp

**Pull request merged**